---
title: "Heart Disease Classification Assignment 6"
author: 
  - name: "Yichen Shen"
    id: "400379853"
  - name: "Yanxi Xiong"
    id: ""
  - name: ""
    id: ""
format: pdf
fontsize: 11pt
linestretch: 1.5
mainfont: "Times New Roman"
geometry: margin=1in
---



Yichen Shen 400379853  


\newpage

In [6]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.feature_selection import SelectKBest, f_classif
from sklearn.metrics import accuracy_score, recall_score
from sklearn.preprocessing import StandardScaler
from sklearn.cluster import KMeans
from sklearn.ensemble import RandomForestClassifier
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.decomposition import PCA


In [7]:
df=pd.read_csv('https://raw.githubusercontent.com/PratheepaJ/datasets/refs/heads/master/ass6-dataset.csv',header=None,encoding='latin1')
columns=['age','sex','cp','trestbps','chol','fbs','restecg','thalach','exang','oldpeak','slope','ca','thal','num']
df.columns=columns
df

age  sex  cp  trestbps  chol  fbs  ...  exang  oldpeak  slope   ca  thal  num
0    age  sex  cp  trestbps  chol  fbs  ...  exang  oldpeak  slope   ca  thal  num
1     63    1   1       145   233    1  ...      0      2.3      3  0.0   6.0    0
2     67    1   4       160   286    0  ...      1      1.5      2  3.0   3.0    2
3     67    1   4       120   229    0  ...      1      2.6      2  2.0   7.0    1
4     37    1   3       130   250    0  ...      0      3.5      3  0.0   3.0    0
..   ...  ...  ..       ...   ...  ...  ...    ...      ...    ...  ...   ...  ...
299   45    1   1       110   264    0  ...      0      1.2      2  0.0   7.0    1
300   68    1   4       144   193    1  ...      0      3.4      2  2.0   7.0    2
301   57    1   4       130   131    0  ...      1      1.2      2  1.0   7.0    3
302   57    0   2       130   236    0  ...      0      0.0      2  1.0   3.0    1
303   38    1   3       138   175    0  ...      0      0.0      1  NaN   3.0    0

[304 rows x 14 columns]

In [3]:
# Q1

This classification problem uses the UCI Heart Disease Dataset to determine whether a patient has heart disease.  
The input includes 14 features such as age, sex, oldpeak and thalach. The target variables indicate heart diseas status, o means no disease and 1 means presence of disease.  
The model is trained to classify new patient cases based on these predictors.

In [10]:
# Q2
df=df.apply(pd.to_numeric,errors='coerce')
numerical_cols=df.select_dtypes(include=[np.number]).columns
scaler=StandardScaler()
df[numerical_cols]=scaler.fit_transform(df[numerical_cols])

# I did the standardized transformation to improve the model performance.

In [11]:
# Q3
print("The data type of the dataset is \n",df.dtypes)
print("\n")
print("Variables in the dataset is \n",df.columns.tolist())
print("\n")
print("Number of observations in the dataset is \n",len(df))


The data type of the dataset is 
 age         float64
sex         float64
cp          float64
trestbps    float64
chol        float64
fbs         float64
restecg     float64
thalach     float64
exang       float64
oldpeak     float64
slope       float64
ca          float64
thal        float64
num         float64
dtype: object


Variables in the dataset is 
 ['age', 'sex', 'cp', 'trestbps', 'chol', 'fbs', 'restecg', 'thalach', 'exang', 'oldpeak', 'slope', 'ca', 'thal', 'num']


Number of observations in the dataset is 
 304


The dataset includes 304 observations with 14 features, where num is the response variable, indicating the presence or absence of heart disease. The remaining 13 variables represent medical features such as age, cholesterol lever.  
All columns are successfully converted to numerical format for modeling purpose, showing float64 types across the dataset. This can help compatibility with machine learning.  
The variables show a diverse range of values. For instance, age ranges from young adults to elderly patients, while chol and thalach span a wide range of clinical readings. This means that the dataset captures a broad spectrum of patient conditions.